File: src/simenv/assets/gltflib/models/extensions/hf_rl_agent.py
from dataclasses import dataclass
from typing import List, Optional

from dataclasses_json import dataclass_json


@dataclass_json
@dataclass
class HFRlAgent:
    """
    A GTLF RL Agent
    """

    color: Optional[List[float]] = None
    height: Optional[float] = None
    move_speed: Optional[float] = None
    turn_speed: Optional[float] = None
    action_name: Optional[str] = None
    action_dist: Optional[str] = None
    camera_width: Optional[int] = None
    camera_height: Optional[int] = None
    available_actions: Optional[List[str]] = None
    reward_functions: Optional[List[str]] = None
    reward_entity1s: Optional[List[str]] = None
    reward_entity2s: Optional[List[str]] = None
    reward_distance_metrics: Optional[List[str]] = None
    reward_scalars: Optional[List[float]] = None
    reward_thresholds: Optional[List[float]] = None
    reward_is_terminals: Optional[List[bool]] = None
    reward_is_collectables: Optional[List[bool]] = None

File: integrations/Unity/simenv-unity/Assets/SimEnv/Runtime/RLAgents/HF_RL_Agent.cs
using System.Collections.Generic;

namespace SimEnv.Agents {
    public class HFRlAgent : IGLTFExtension {
        //public string name = "";
        public float[] color;
        public float height = 1f;
        public float move_speed = 1f;
        public float turn_speed = 1f;
        public string action_name = "";
        public string action_dist = "";

        public List<string> available_actions = new List<string>();

        // reward function definitions
        public List<string> reward_functions = new List<string>();
        public List<string> reward_entity1s = new List<string>();
        public List<string> reward_entity2s = new List<string>();
        public List<string> reward_distance_metrics = new List<string>();
        public List<float> reward_scalars = new List<float>();
        public List<float> reward_thresholds = new List<float>();
        public List<bool> reward_is_terminals = new List<bool>();
        public List<bool> reward_is_collectables = new List<bool>();

        public void Initialize(Node node) {
            new Agent(node, this);
        }
    }
}